lib/engine/step/g_1870/price_protection.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative '../buy_sell_par_shares'
+
+module Engine
+  module Step
+    module G1870
+      class PriceProtection < Base
+        def actions(entity)
+          return [] if !entity.player? || entity != current_entity
+
+          actions = []
+          actions << 'buy_shares' if can_buy?(entity, protecting_bundle)
+          actions << 'pass' if actions.any?
+          actions
+        end
+
+        def description
+          'Price protect shares'
+        end
+
+        def round_state
+          { sell_queue: [] }
+        end
+
+        def purchasable_companies(_entity = nil)
+          []
+        end
+
+        def protecting_bundle
+          @round.sell_queue.dig(0)
+        end
+
+        def can_sell?
+          false
+        end
+
+        def can_buy?(entity, bundle)
+          return unless bundle
```

Comment:
return unless bundle&.buyable

Reply:
This was copy-pasted from `buy_sell_par_shares.rb`. I'm sneaking in a change in there as well

Reply:
you can still clean it up

Reply:
what did you change?

Reply:
The code you suggested I edited was copy-pasted from another step. I changed it in my new step and in the original step.